metabase/metabase
# De-crazy the expression editor function helper popover

Closes https://github.com/metabase/metabase/issues/28668, check it for full details

### Description

We want to improve expression editor helper popover to make it less overwhelming and more clean and helpful.
To achieve that we want to change it's design and structure.

<table>
<tr>
<th> Before: </th>
<th> After: </th>
</tr>
<tr>
<td>

<img width="350" alt="Screenshot 2023-02-27 at 19 34 09" src="https://user-images.githubusercontent.com/7983744/221607763-7f4cd944-00e9-4b0e-9057-ce1487ce7bd1.png">

</td>
<td>

<img width="350" alt="Screenshot 2023-02-27 at 19 41 14" src="https://user-images.githubusercontent.com/7983744/221609926-ef52ac72-0e23-4190-b96b-53f5f4f5b81a.png">


</td>
</tr>
</table>

### How to verify

1. Start creating a new Question: click "New" button in the top bar -> Question.
2. Click "Custom column" in the Notebook.
3. Appeared expression editor should have new design and experience.

### Demo

https://user-images.githubusercontent.com/7983744/223769602-b2deaf75-3e47-4891-92c6-f76644abf793.mov


### Checklist

- [x] [Refactored ExpressionWidget, ClauseStep components to TS](https://github.com/metabase/metabase/pull/28687)
- [x] [Change expression editor title, Implement info tooltip](https://github.com/metabase/metabase/pull/28776) 
- [x] [Change popover structure to match proposed design](https://github.com/metabase/metabase/pull/28920)
- [x] [Add interactivity for popover parameters](https://github.com/metabase/metabase/pull/28966)


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/28678)
